src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,592 @@
+"""Calculate incident angles using a sim_telarray PSF-style run.
+
+Parses the imaging list (``.lis``) produced by sim_telarray_debug_trace and uses
+Angle of incidence at focal surface, with respect to the optical axis [deg],
+Angle of incidence on(to) primary mirror [deg], and
+Angle of incidence on(to) secondary mirror [deg] (if available).
+"""
+
+import logging
+import re
+import subprocess
+from pathlib import Path
+
+import astropy.units as u
+from astropy.table import QTable
+
+from simtools.model.model_utils import initialize_simulation_models
+
+
+class IncidentAnglesCalculator:
+    """Run a PSF-style sim_telarray job and compute incident angles at mirrors or focal surfaces.
+
+    Parameters
+    ----------
+    simtel_path : str or pathlib.Path
+        Path to the sim_telarray installation directory (containing ``sim_telarray/bin``).
+    db_config : dict or str or None
+        Database configuration passed to ``initialize_simulation_models``.
+    config_data : dict
+        Simulation configuration (e.g. ``site``, ``telescope``, ``model_version``,
+        ``zenith_angle``, ``off_axis_angle``, ``source_distance``, ``number_of_rays``).
+    output_dir : str or pathlib.Path
+        Output directory where logs, scripts, photons files and results are written.
+    label : str, optional
+        Label used to name outputs; defaults to ``incident_angles_<telescope>`` when omitted.
+    perfect_mirror : bool, default False
+        If True, enforce perfect mirror settings (no random errors in sim_telarray options).
+    overwrite_rdna : bool, default False
+        If True and ``mirror_reflection_random_angle`` is None, force that parameter to 0.
+    mirror_reflection_random_angle : float, optional
+        Mirror reflection random angle (sets sim_telarray ``mirror_reflection_random_angle``).
+    algn : float, optional
+        Mirror alignment randomization amplitude used for horizontal/vertical settings.
+    test : bool, default False
+        If True, reduce the number of rays for a faster run (keeps outputs compatible).
+    calculate_primary_secondary_angles : bool, default True
+        If True, parse and store primary/secondary mirror incidence angles from the imaging list.
+    """
+
+    def __init__(
+        self,
+        simtel_path,
+        db_config,
+        config_data,
+        output_dir,
+        label=None,
+        perfect_mirror=False,
+        overwrite_rdna=False,
+        mirror_reflection_random_angle=None,
+        algn=None,
+        test=False,
```

Comment:
Suggest to remove test, as it is confusing.

e.g. we define an integration test with 1000 photons. But setting test true, means that a different number of photons are generated. As we have nowadays configuration files for tests, there is no need for this parameter.